hybrid/SampleApps/NoteSync/src/com/salesforce/samples/notesync/ContentSoqlSyncTarget.java

```diff
@@ -0,0 +1,257 @@
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Target for sync defined by a SOQL query
+ */
+public class ContentSoqlSyncTarget extends SoqlSyncTarget {
+
+    private static final String REQUEST_TEMPLATE = "<?xml version=\"1.0\"?>" +
+            "<se:Envelope xmlns:se=\"http://schemas.xmlsoap.org/soap/envelope/\">" +
+            "<se:Header xmlns:sfns=\"urn:partner.soap.sforce.com\">" +
+            "<sfns:SessionHeader><sessionId>%s</sessionId></sfns:SessionHeader>" +
+            "</se:Header>" +
+            "<se:Body>%s</se:Body>" +
+            "</se:Envelope>";
+
+    private static final String QUERY_TEMPLATE = "<query xmlns=\"urn:partner.soap.sforce.com\" xmlns:ns1=\"sobject.partner.soap.sforce.com\">" +
+            "<queryString>%s</queryString></query>";
+
+
+    private static final String QUERY_MORE_TEMPLATE = "<queryMore xmlns=\"urn:partner.soap.sforce.com\" xmlns:ns1=\"sobject.partner.soap.sforce.com\">\n"+
+            "        <queryLocator>%s</queryLocator>\n"+
+            "    </queryMore>";
+
+    public static final String RESULT = "result";
+    public static final String RECORDS = "records";
+    public static final String SF = "sf:";
+    public static final String QUERY_LOCATOR = "queryLocator";
+    public static final String SIZE = "size";
+    public static final String DONE = "done";
+
+    private String queryLocator;
+
+    /**
+     * Build SyncTarget from json
+     * @param target as json
+     * @return
+     * @throws JSONException
+     */
+    public static SyncTarget fromJSON(JSONObject target) throws JSONException {
+        if (target == null)
+            return null;
+
+        String query = target.getString(QUERY);
+        return new ContentSoqlSyncTarget(query);
+    }
+
+    /**
+     * Build SyncTarget for soql target
+     * @param soql
+     * @return
+     */
+    public static ContentSoqlSyncTarget targetForSOQLSyncDown(String soql) {
+        return new ContentSoqlSyncTarget(soql);
+    }
+
+    /**
+     * Private constructor
+     * @param query
+     */
+    public ContentSoqlSyncTarget(String query) {
+        super(query);
+        this.queryType = QueryType.custom;
+    }
+
+    /**
+     * @return json representation of target
+     * @throws JSONException
+     */
+    public JSONObject asJSON() throws JSONException {
+        JSONObject target = super.asJSON();
+        target.put(ANDROID_IMPL, getClass().getName());
+        return target;
+    }
+
+    @Override
+    public JSONArray startFetch(SyncManager syncManager, long maxTimeStamp) throws IOException, JSONException {
+        String queryToRun = maxTimeStamp > 0 ? SoqlSyncTarget.addFilterForReSync(getQuery(), maxTimeStamp) : getQuery();
+        syncManager.getRestClient().sendSync(RestRequest.getRequestForResources(syncManager.apiVersion)); // cheap call to refresh session
```

Comment:
Soap expects the session id in the body of the post. So we are making a cheap call to make sure we have a valid session id first.